src/app/libp2p_helper/src/codanet.go

```diff
@@ -118,9 +137,57 @@ func (cm *CodaConnectionManager) ClosedStream(net network.Network, stream networ
 	cm.p2pManager.Notifee().ClosedStream(net, stream)
 }
 func (cm *CodaConnectionManager) Connected(net network.Network, c network.Conn) {
+	logger.Debugf("%s connected to %s", c.LocalPeer(), c.RemotePeer())
 	cm.OnConnect(net, c)
 	cm.p2pManager.Notifee().Connected(net, c)
+
+	if !cm.minaPeerExchange {
+		return
+	}
+
+	info := cm.GetInfo()
+	if len(net.Peers()) <= info.HighWater {
+		return
+	}
+	
+	logger.Debugf("node=%s disconnecting from peer=%s; max peers=%d peercount=%d", c.LocalPeer(), c.RemotePeer(), info.HighWater, len(net.Peers()))
+
+	// select random subset of our peers to send over, then disconnect
+	if cm.getRandomPeers == nil {
+		logger.Error("getRandomPeers function not set")
+		return
+	}
+
+	peers := cm.getRandomPeers(numPeersToExchange, c.RemotePeer())
```

Comment:
Actually I think instead of passing `numPeersToExchange` here, I would pass `info.LowWater`, since that's anyway the minimum number of connections the node on the other end is trying to maintain.